src/base/utils/misc.cpp

```diff
@@ -80,21 +82,26 @@ namespace
     // see http://en.wikipedia.org/wiki/Kilobyte
     // value must be given in bytes
     // to send numbers instead of strings with suffixes
-    bool splitToFriendlyUnit(const qint64 sizeInBytes, qreal &val, Utils::Misc::SizeUnit &unit)
+    struct SplitToFriendlyUnitResult
     {
-        if (sizeInBytes < 0) return false;
+        qreal value;
+        Utils::Misc::SizeUnit unit;
+    };
+
+    std::optional<SplitToFriendlyUnitResult> splitToFriendlyUnit(const qint64 bytes)
```

Comment:
Is negative bytes valid parameter?
Currently it is handled as valid but some kind of "special" value. Seems incorrect to me.
Although this behavior is not introduced by the current PR, so you can not touch it.

Reply:
>Currently it is handled as valid but some kind of "special" value. Seems incorrect to me.

You meant the "Unknown" in `Utils::Misc::friendlyUnit()` right?

>Although this behavior is not introduced by the current PR, so you can not touch it.

I better leave it for others to solve.